### PR TITLE
add reportobligation date

### PR DESCRIPTION
**Pull request by fabiangotzen**

### CODE REVIEW

File: infrastructure/src/main/resources/mails/i18n/mail-messages.properties [hunk 1]
```diff
@@ -4,15 +4,19 @@ mail.jobAd.created.header=Guten Tag
...
+mail.jobad.created.body.reporting.obligation.date.text2=anderweitig ausgeschrieben werden.
 mail.jobAd.created.body.paragraph3=Falls Sie in der Zwischenzeit Fragen haben, zögern Sie nicht, Ihr zuständiges RAV zu kontaktieren:
 mail.jobAd.created.body.paragraph4=Über diesen Link können Sie die Publikation Ihrer Vakanz duplizieren oder beenden:
 mail.jobAd.created.body.addressblock.tel=Tel.:
 mail.jobAd.created.body.addressblock.fax=Fax:
 mail.jobAd.created.body.addressblock.email=Email:
 
 mail.jobAd.refined.header=Guten Tag
-mail.jobAd.refined.body.paragraph1=Gerne teilen wir Ihnen mit, dass Ihr Stelleninserat vom Regionalen Arbeitsvermittlungszentrum (RAV) validiert und im Job-Room sowie im System der öffentliche Arbeitsvermittlung AVAM aufgeschaltet wurde.
+mail.jobAd.refined.body.paragraph1=Gerne teilen wir Ihnen mit, dass Ishr Stelleninserat vom Regionalen Arbeitsvermittlungszentrum (RAV) validiert und im Job-Room sowie im System der öffentliche Arbeitsvermittlung AVAM aufgeschaltet wurde.
```
Comment (pavax) [inline infrastructure/src/main/resources/mails/i18n/mail-messages.properties]: no no no

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java [hunk 2]
```diff
@@ -71,6 +73,7 @@ void onCreated(JobAdvertisementCreatedEvent event) {
         Map<String, Object> variables = new HashMap<>();
         variables.put("jobAdvertisement", jobAdvertisement);
         variables.put("jobCenter", jobCenter);
+        variables.put("reportingObligationEndDate",(jobAdvertisement.getReportingObligationEndDate() != null) ? jobAdvertisement.getReportingObligationEndDate().format(DATE_FORMATTER) : null);
```
Comment (pavax) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java]: extract method

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java [hunk 3]
```diff
@@ -101,6 +104,7 @@ void onRefined(JobAdvertisementRefinedEvent event) {
...
```
Comment (pavax) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java]: is this the same code such as the above code?

File: infrastructure/src/main/resources/mails/i18n/mail-messages.properties [hunk 4]
```diff
@@ -4,15 +4,19 @@ mail.jobAd.created.header=Guten Tag
...
 mail.jobAd.created.body.paragraph2=Ihre Stellenmeldung wurde an das Regionale Arbeitsvermittlungszentrum (RAV) zur Prüfung weitergeleitet. Bei allfälligen Fragen werden Sie vom RAV kontaktiert. Sie werden per E-Mail informiert, sobald die Stelle aufgeschaltet wurde.
 mail.jobAd.created.body.reporting.obligation=Achtung Stellenmeldepflicht - Die automatische Erkennung unseres Systems hat festgestellt, dass die Stelle der Stellenmeldepflicht gemäss Art. 21a Ausländergesetz sowie Art. 53b Arbeitsvermittlungsverordnung unterliegt. Der Zugriff auf die Informationen dieser Stelle wird während einer Frist von 5 Arbeitstagen auf Personen, die bei der öffentlichen Arbeitsvermittlung als Stellensuchende angemeldet sind, beschränkt. Sie dürfen diese Stelle erst nach Ablauf dieser Frist anderweitig ausschreiben. Genaue Angaben zu dieser Frist erhalten Sie zusammen mit der Bestätigung, dass die Stelle im Job-Room aufgeschaltet ist.
+mail.jobad.created.body.reporting.obligation.date.text1=Diese Stelle darf erst ab
+mail.jobad.created.body.reporting.obligation.date.text2=anderweitig ausgeschrieben werden.
```
Comment (tonim42) [inline infrastructure/src/main/resources/mails/i18n/mail-messages.properties]: Nicht bei jobAd.created - erst nach Publikation!

File: infrastructure/src/main/resources/mails/i18n/mail-messages_de.properties [hunk 5]
```diff
@@ -2,6 +2,8 @@ mail.jobAd.created.header=Guten Tag
...
 mail.jobAd.created.body.paragraph2=Ihre Stellenmeldung wurde an das Regionale Arbeitsvermittlungszentrum (RAV) zur Prüfung weitergeleitet. Bei allfälligen Fragen werden Sie vom RAV kontaktiert. Sie werden per E-Mail informiert, sobald die Stelle aufgeschaltet wurde.
 mail.jobAd.created.body.reporting.obligation=Achtung Stellenmeldepflicht - Die automatische Erkennung unseres Systems hat festgestellt, dass die Stelle der Stellenmeldepflicht gemäss Art. 21a Ausländergesetz sowie Art. 53b Arbeitsvermittlungsverordnung unterliegt. Der Zugriff auf die Informationen dieser Stelle wird während einer Frist von 5 Arbeitstagen auf Personen, die bei der öffentlichen Arbeitsvermittlung als Stellensuchende angemeldet sind, beschränkt. Sie dürfen diese Stelle erst nach Ablauf dieser Frist anderweitig ausschreiben. Genaue Angaben zu dieser Frist erhalten Sie zusammen mit der Bestätigung, dass die Stelle im Job-Room aufgeschaltet ist.
+mail.jobad.created.body.reporting.obligation.date.text1=Diese Stelle darf erst ab
+mail.jobad.created.body.reporting.obligation.date.text2=anderweitig ausgeschrieben werden.
```
Comment (tonim42) [inline infrastructure/src/main/resources/mails/i18n/mail-messages_de.properties]: ditto

File: infrastructure/src/main/resources/mails/i18n/mail-messages.properties [hunk 6]
```diff
@@ -4,6 +4,8 @@ mail.jobAd.created.header=Guten Tag
 mail.jobAd.created.body.paragraph1=Besten Dank für Ihre Stellenmeldung, es freut uns, dass Sie die Plattform Job-Room benutzen.
 mail.jobAd.created.body.paragraph2=Ihre Stellenmeldung wurde an das Regionale Arbeitsvermittlungszentrum (RAV) zur Prüfung weitergeleitet. Bei allfälligen Fragen werden Sie vom RAV kontaktiert. Sie werden per E-Mail informiert, sobald die Stelle aufgeschaltet wurde.
 mail.jobAd.created.body.reporting.obligation=Achtung Stellenmeldepflicht - Die automatische Erkennung unseres Systems hat festgestellt, dass die Stelle der Stellenmeldepflicht gemäss Art. 21a Ausländergesetz sowie Art. 53b Arbeitsvermittlungsverordnung unterliegt. Der Zugriff auf die Informationen dieser Stelle wird während einer Frist von 5 Arbeitstagen auf Personen, die bei der öffentlichen Arbeitsvermittlung als Stellensuchende angemeldet sind, beschränkt. Sie dürfen diese Stelle erst nach Ablauf dieser Frist anderweitig ausschreiben. Genaue Angaben zu dieser Frist erhalten Sie zusammen mit der Bestätigung, dass die Stelle im Job-Room aufgeschaltet ist.
+mail.jobad.created.body.reporting.obligation.date.text1=Diese Stelle darf erst ab
```
Comment (pavax) [inline infrastructure/src/main/resources/mails/i18n/mail-messages.properties]: you could use variables such use: {0}

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java [hunk 7]
```diff
@@ -26,6 +27,7 @@
 @Component
 public class JobAdvertisementMailEventListener {
 
+    public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
```
Comment (pavax) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java]: why public

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java [hunk 8]
```diff
@@ -186,4 +189,8 @@ private JobAdvertisement getJobAdvertisement(JobAdvertisementId jobAdvertisement
         return jobAdvertisement.orElseThrow(() -> new AggregateNotFoundException(JobAdvertisement.class, jobAdvertisementId.getValue()));
     }
 
+    private String formattedReportingObligationEndDateOrNull(JobAdvertisement jobAdvertisement) {
```
Comment (pavax) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java]: parameter should be a date -> rename to nullSafeFormatLocalDate

**Pull request merged**